File: scripts/test_glm53.mjs
const apiKey = process.env.NVIDIA_API_KEY;
if (!apiKey) { console.error("Set NVIDIA_API_KEY in your shell (never commit it)."); process.exit(1); }
const url = "https://integrate.api.nvidia.com/v1/chat/completions";

console.log("Sending request to NVIDIA NIM API for z-ai/glm-5.3...");
const startTime = Date.now();

try {
  const res = await fetch(url, {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
      "Authorization": `Bearer ${apiKey}`,
    },
    body: JSON.stringify({
      model: "z-ai/glm-5.3",
      messages: [
        { role: "system", content: "You are a helpful assistant." },
        { role: "user", content: "Which number is larger, 9.11 or 9.8?" }
      ],
      temperature: 0.5,
      top_p: 1,
      max_tokens: 1024,
      stream: false
    })
  });

  const duration = ((Date.now() - startTime) / 1000).toFixed(2);
  console.log(`Status: ${res.status} ${res.statusText} (${duration}s)`);
  
  const data = await res.json();
  console.log("Response data:", JSON.stringify(data, null, 2));
} catch (err) {
  console.error("Fetch error:", err);
}

File: scripts/test_glm53.py
import sys
from openai import OpenAI

try:
    client = OpenAI(
        base_url="https://integrate.api.nvidia.com/v1",
        api_key="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"
    )

    print("Connecting to NVIDIA API for model z-ai/glm-5.3...")
    completion = client.chat.completions.create(
        model="z-ai/glm-5.3",
        messages=[
            {"role": "system", "content": "You are a helpful assistant."},
            {"role": "user", "content": "Which number is larger, 9.11 or 9.8?"}
        ],
        temperature=0.5,
        top_p=1,
        max_tokens=1024,
        stream=False
    )

    print("Response received successfully:")
    print(completion.choices[0].message.content)
except Exception as e:
    print(f"Error: {e}", file=sys.stderr)
    sys.exit(1)
